### PR TITLE
DLUHC-201 add configurable page for timetable stages

### PR DESCRIPTION
**Pull request by samornsby**
Adds a new input page for the Timetable Creation Form which allows us to enter the stages of a plan e.g. `Scoping of the New Local Plan`, `Gateway 1. Check-point`, `Call for evidence`, etc.

- Added  `StagePage` which can configure a stage for the timetable. Since each stage is an identical data structure and the stages appear to be fixed based on UX, I've added props to this component so it can be easily reused.
- Updated `DateInput` to allow entering a data of just month and year. It might be worth a completely different input to easily validate this data compared to a full date, I'm not too sure yet without wiring up all the data.
- Added a type for the pages to allow arbitrary props to be passed into each component.
- Configure two stages on the form, I'll add the full list once it's all connected to the state. Two should be enough to prove functionality for now.

Again styling to match UX will be done in a separate PR:
image

### CODE REVIEW

File: dluhc-component-library/src/components/timetableForm/types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { JSXInternal } from "node_modules/preact/src/jsx";
+
+export interface TimetablePage {
+  id: string;
+  component: (props?: any) => JSXInternal.Element;
+  props?: any;
```
[the remark this answers is not in the document]
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/types.ts:6]: I thought about doing something like this, but I think it then means we have to pass a type in when we use it `TimetablePage<Something>` I'm not sure what that `Something` type would be be, would it be `any`? If I make it more specific it then complains that the type might be undefined for the components with props.

File: dluhc-component-library/src/components/timetableForm/components/StagePage.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  Textarea,
+} from "src/components/formComponents";
+import { RadioOption } from "src/components/formComponents/types";
+
+interface StagePageProps {
+  stageName: string;
+}
+
+const PROGRESS_OPTIONS: ReadonlyArray<RadioOption<string>> = [
+  { label: "Not Started", value: "notStarted" },
```
[the remark this answers is not in the document]
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/components/StagePage.tsx:13]: Yeah can do, if they're used across two components is there a good place to put them? A new folder for common types at a higher level maybe?

[the remark this answers is not in the document]
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/components/StagePage.tsx]: I've added a `models/timetable` folder with `constants` and `types`

File: dluhc-component-library/src/components/timetableForm/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/types.ts]: Made that type change and have moved to component rendering into the memo to conditionally apply the props if they exists.

**Review by amjones-scottlogic — APPROVED** [review]
LGTM

**Pull request merged**